variation-commons-mongodb/src/test/java/uk/ac/ebi/eva/commons/mongodb/writers/VariantMongoWriterTest.java

```diff
@@ -0,0 +1,317 @@
+
+        DBCollection dbCollection = mongoOperations.getCollection(collectionName);
+        assertEquals(1, dbCollection.count());
+        final DBObject storedVariant = dbCollection.findOne();
+        assertNull(storedVariant.get(IDS_FIELD));
+    }
+
+    @Test
+    public void sourceEntryShouldBeWritten() throws Exception {
+        final String chromosome = "12";
+        final long start = 3;
+        final long end = 4;
+        final String reference = "A";
+        final String alternate = "T";
+        final String fileId = "fileId";
+        final String studyId = "studyId";
+        Variant variant = buildVariantWithSourceEntry(chromosome, start, end, reference, alternate, fileId, studyId);
+
+        VariantMongoWriter variantMongoWriter = new VariantMongoWriter(collectionName, mongoOperations, false, true);
+        variantMongoWriter.write(Collections.singletonList(variant));
+
+        DBCollection dbCollection = mongoOperations.getCollection(collectionName);
+        assertEquals(1, dbCollection.count());
+        final DBObject storedVariant = dbCollection.findOne();
+        final BasicDBList variantSources = (BasicDBList) storedVariant.get(FILES_FIELD);
+        assertNotNull(variantSources);
+        assertFalse(variantSources.isEmpty());
+        DBObject sourceEntry = (DBObject) variantSources.get(0);
+        assertEquals(fileId, sourceEntry.get(FILEID_FIELD));
+        assertEquals(studyId, sourceEntry.get(STUDYID_FIELD));
+        assertNotNull(sourceEntry.get(ALTERNATES_FIELD));
+        assertNotNull(sourceEntry.get(ATTRIBUTES_FIELD));
+        assertNotNull(sourceEntry.get(FORMAT_FIELD));
+        assertNotNull(sourceEntry.get(SAMPLES_FIELD));
+    }
+
+    @Test
+    public void sourceEntryShouldBeAdded() throws Exception {
+        Variant variant = buildVariantWithSourceEntry("1", 1, 2, "A", "T", "fileId", "previousStudy");
+        Variant newVariant = buildVariantWithSourceEntry("1", 1, 2, "A", "T", "fileId", "dbsnpStudy");
+
+        VariantMongoWriter variantMongoWriter = new VariantMongoWriter(collectionName, mongoOperations, false, false);
+        variantMongoWriter.write(Collections.singletonList(variant));
+
+        variantMongoWriter.write(Collections.singletonList(newVariant));
+        DBCollection dbCollection = mongoOperations.getCollection(collectionName);
+        assertEquals(1, dbCollection.count());
+        final DBObject storedVariant = dbCollection.findOne();
+        assertEquals(2, ((BasicDBList) storedVariant.get(FILES_FIELD)).size());
+    }
+
+    private Variant buildVariant(String chromosome, long start, long end, String reference, String alternate,
+                                 String fileId, String studyId) {
+        return buildVariant(chromosome, start, end, reference, alternate, new VariantSourceEntry(fileId, studyId));
+    }
+
+    private Variant buildVariantWithSourceEntry(String chromosome, long start, long end, String reference, String alternate,
```

Comment:
The previous method is also creating a variant with a "Source Entry". Can we rename this method to something like "buildVariantWithSampleData", "buildVariantWithGenotypes" or similar?